### PR TITLE
Add pages + ironman filter to lb minigames

### PR DESCRIPTION
**Pull request by TastyPumPum**
- [x] I have tested all my changes thoroughly.

----

| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit 6ebc5640acab8c499569689603a804b3a68fb04e  | 
|--------|--------|

### Summary:
Added pagination and ironman filter to minigames leaderboard in `src/mahoji/commands/leaderboard.ts`.

**Key points**:
- Added pagination to minigames leaderboard in `src/mahoji/commands/leaderboard.ts`.
- Introduced `ironmanOnly` parameter to `minigamesLb` function for filtering ironman-only results.
- Updated SQL queries to include ironman filter and pagination.
- Modified `doMenuWrapper` and `doMenu` functions to support new pagination feature.
- Adjusted command options in `leaderboardCommand` to include `ironmanOnlyOption`.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

**Comment by TastyPumPum** [comment]
@ellipsis-dev review this pr

File: src/mahoji/commands/leaderboard.ts [hunk 1]
```diff
@@ -291,27 +316,56 @@ async function minigamesLb(interaction: ChatInputCommandInteraction, user: MUser
...
+			user,
+			channelID,
+			users: championsCompletions.map(c => ({
+				id: c.id,
+				score: c.amount
+			})),
+			title: 'Champions Challenge Leaderboard',
+			ironmanOnly
+		});
+	}
```
Comment (gc) [inline src/mahoji/commands/leaderboard.ts:344]: Why is this change needed?

Comment (TastyPumPum) [reply src/mahoji/commands/leaderboard.ts]: Have removed now and tested
